src/main/java/uk/gov/pay/api/model/search/card/SearchRefunds.java

```diff
@@ -0,0 +1,131 @@
+        Response connectorResponse = client
+                .target(url)
+                .request()
+                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                .get();
+        logger.info("response from connector for refunds search: " + connectorResponse);
+        if (connectorResponse.getStatus() == SC_OK) {
+            return processResponse(account.getAccountId(), connectorResponse);
+        }
+        throw new SearchRefundsException(connectorResponse);
+    }
+
+    private Response processResponse(String accountId, Response connectorResponse) {
+        try {
+            JsonNode responseJson = connectorResponse.readEntity(JsonNode.class);
+            TypeReference<SearchRefundsResponse> typeRef = new TypeReference<SearchRefundsResponse>() {
+            };
+            SearchRefundsResponse searchResponse = objectMapper.readValue(responseJson.traverse(), typeRef);
+            List<RefundForSearchRefundsResult> results = searchResponse.getRefunds()
+                    .stream()
+                    .map(refund -> RefundForSearchRefundsResult.valueOf(refund))
+                    .collect(Collectors.toList());
+
+            HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation
+                    .builder()
+                    .addProperty("results", results);
+            return Response.ok().entity(decoratePagination(Long.valueOf(accountId), halRepresentation, searchResponse)
+                    .build().toString())
+                    .build();
+        } catch (IOException | ProcessingException ex) {
+            throw new SearchRefundsException(ex);
+        }
+    }
+
+    protected HalRepresentation.HalRepresentationBuilder decoratePagination(Long accountId,
+                                                                            HalRepresentation.HalRepresentationBuilder halRepresentationBuilder,
+                                                                            SearchRefundsResponse pagination) {
+        try {
+            halRepresentationBuilder
+                    .addProperty("count", pagination.getCount())
+                    .addProperty("total", pagination.getTotal())
+                    .addProperty("page", pagination.getPage());
+            addLink(halRepresentationBuilder, "self", transformIntoPublicUri(
+                    baseUrl,
+                    pagination.getLinks().getSelfPage(),
+                    accountId
+            ));
+        } catch (URISyntaxException ex) {
+            throw new SearchRefundsException(ex);
+        }
+        return halRepresentationBuilder;
+    }
+
+    private void addLink(HalRepresentation.HalRepresentationBuilder halRepresentationBuilder, String name, URI uri) {
+        if (uri != null) {
+            halRepresentationBuilder.addLink(name, uri);
+        }
+    }
+
+    private URI transformIntoPublicUri(String baseUrl, Link link, Long accountId) throws URISyntaxException {
```

Comment:
same as above, this duplicates a method in https://github.com/alphagov/pay-publicapi/blob/f16da88549f0bd4a8f2063a06876f8e78cec154f/src/main/java/uk/gov/pay/api/model/search/SearchPaymentsBase.java

We might want payents and refund to inherit from a common base class, otherwise we are going to duplicate lots of 💩